Check remote tags before deciding to do a fetch #42329

asyncmind0 opened the pull request:
### What does this PR do?
Check for new tags in remote repo, before deciding to fetch in git.latest
### What issues does this PR fix or reference?
Fixes  #42329 
### Previous Behavior
new tags in remote repo are ignored

### New Behavior
new tags in remote repo are fetched

### Tests written?

No

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

BenjaminSchubert committed:
udpated docker_container doc for selinux

The selinux flags for volumes where missing.

terminalmage committed:
Introduce "priority" feature for pyobjects

When there are multiple matches (such as when both an os_family and os
grain both match the minion), the order in which they are processed is
determined by the iteration order of the class' __dict__ attribute.
Since this would be different in different versions of Python, behavior
would be inconsistent depending on which version of Python is being used.

This adds a priority attribute to the Map subclass which, if present,
will re-order the match groups such that the results will be predictable
irrespective of Python version.

BenjaminSchubert committed:
updated documentation for docker bind selinux

skizunov committed:
PY3: Fix importlib usage to use fullname

When a module wasn't loaded using fullname, it wouldn't be able to
find the module by function since `<funcname>.__module__` wouldn't
match what is in `syspaths.modules`. Due to this, calling module functions
on the command line wouldn't work. Fixed this issue. Also optimized usage
of importlib by switching to the `importlib.util.spec_from_file_location`
API.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

Commit:
Merge pull request #41952 from meaksh/2017.7-suse-leftovers-part-two

Fix "zypp_plugins" unit test for Python 3

Commit:
Merge pull request #41913 from twangboy/win_servermanager

Win servermanager

Commit:
Merge pull request #41960 from skizunov/develop3

PY3: Fix importlib usage to use fullname

Commit:
Merge pull request #41939 from cachedout/mdadm_py3_test

Gate assert check behind mock ver

skizunov committed:
PY3: Don't use importlib.util.module_from_spec in PY3.4

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

Commit:
Merge pull request #41959 from DSRCorporation/bugs/minor_dict_diff_enhance

Minor dict diff enhance in tests.

twangboy committed:
Fix unit tests for ssh

skizunov committed:
Implement copy/deepcopy for ContextDict

There is a use case that causes this error:

```
[WARNING ] The minion function caused an exception: can't pickle
_thread._local objects
```

It happens when `salt.modules.state._get_opts` is invoked and it is
running this line of code:

```
opts = copy.deepcopy(__opts__)
```

It occurs because the `grains` in `__opts__` happens to be a
`NamespacedDictWrapper` with the internal `self.__dict` happening
to be a `ContextDict`. Implementing deepcopy for `ContextDict` fixes
this error.

Signed-off-by: Sergey Kizunov <sergey.kizunov@ni.com>

twangboy committed:
Fix paths

twangboy committed:
Fix paths, skip impossible

Commit:
Properly handle declaration order and grains match in pyobjects.Map.

garethgreenaway committed:
Merge pull request #41966 from twangboy/win_unit_test_ssh

Fix `unit.modules.test_ssh` for Windows

garethgreenaway committed:
Merge pull request #41967 from twangboy/win_unit_test_api

Fix `unit.config.test_api` for Windows

garethgreenaway committed:
Merge pull request #41969 from twangboy/win_unit_test_config

Fix `unit.config.test_config` for Windows

Ch3LL committed:
[develop] Bump latest release version to 2016.11.6

Commit:
Merge pull request #41958 from DSRCorporation/bugs/py3_test_map

Pyobjects Map: prioritize grains by declaration order.

Commit:
Merge pull request #41963 from skizunov/develop3

PY3: Don't use importlib.util.module_from_spec in PY3.4

Commit:
Merge pull request #41979 from Ch3LL/11.6_develop

[develop] Bump latest release version to 2016.11.6

Commit:
Merge pull request #41970 from skizunov/develop4

Implement copy/deepcopy for ContextDict

Commit:
Merge branch '2016.11' into '2017.7'

Conflicts:
  - doc/ref/configuration/master.rst
  - tests/integration/states/test_archive.py
  - tests/unit/transport/test_tcp.py
  - tests/unit/transport/test_zeromq.py

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - salt/modules/state.py
  - salt/modules/yumpkg.py
  - salt/modules/zypper.py
  - salt/states/pkg.py
  - salt/version.py
  - tests/unit/config/test_config.py
  - tests/unit/modules/test_zypper.py

Commit:
Rename transport_test file to test_transport and update imports for pytest

terminalmage committed:
Update test_map to test declarative ordering

Commit:
Lint: Remove extra blank line

terminalmage committed:
Add test for pyobjects priority

twangboy committed:
Fixes problem with path

Commit:
Merge pull request #41989 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

terminalmage committed:
Merge remote-tracking branch 'upstream/2016.11' into merge-2017.7

Conflicts:
- doc/_themes/saltstack2/layout.html
- salt/auth/__init__.py
- salt/utils/schedule.py

Commit:
Merge pull request #41951 from terminalmage/pyobjects-priority

Introduce "priority" feature for pyobjects

Commit:
Merge pull request #42000 from terminalmage/merge-2017.7

Merge 2016.11 into 2017.7 Merge remote-tracking branch 'upstream/2016.11' into merge-2017.7

Commit:
Merge pull request #41990 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

terminalmage committed:
Merge remote-tracking branch 'upstream/2017.7' into merge-develop

BenjaminSchubert committed:
fixing bad documentation

mcalmer committed:
Fix serialize error when requesting a node

vutny committed:
[2017.7] Fix crontab-based Master/Minion scheduled jobs

abulford committed:
Fix dockerng.network_* ignoring of tests=True

Fixes #41976

Commit:
add module and state to help create txt docs from highstate data

Commit:
Merge pull request #42001 from terminalmage/merge-develop

Merge 2017.7 into develop

terminalmage committed:
Clarify the binds configuration when selinux context is present

terminalmage committed:
Express multiple examples as an RST list

garethgreenaway committed:
Merge pull request #42004 from vutny/2017-7-fix-croniter-scheduled-jobs

[2017.7] Fix crontab-based Master/Minion scheduled jobs

Commit:
Merge pull request #42002 from mcalmer/2017.7-fix-kubernetes-serializing

Fix serialize error when requesting a node

mirceaulinic committed:
Out execution module

Commit:
Merge pull request #41992 from twangboy/win_unit_test_archive

Fix `unit.modules.test_archive` on Windows

mirceaulinic committed:
Out mod autodoc

mirceaulinic committed:
rst not md

gtmanfred committed:
Revert "Fix documentation for docker login function in pre-nitrogen release branches"

This reverts commit dfbbeb5946b473b15f62488913749919452dedbc.

Commit:
Merge pull request #42016 from terminalmage/pr-41929

Reopen PR #41929 against 2017.7

terminalmage committed:
Add the new docker state modules to the toctree

terminalmage committed:
salt.modules.dockermod: ensure that json.loads is fed a string

json.loads can only handle bystrings in 3.x in version 3.6 and later.

terminalmage committed:
Move salt.utils.files.mkstemp temporarily back to salt.utils

Old code from earlier branches may cause problems when merging forward,
this moves the function back to its old location and we will work on
getting it permanently renamed in develop.

Commit:
Merge pull request #42019 from gtmanfred/docker

Revert docker documentation from pre nitrogen release that was merged forward.

twangboy committed:
Set file_roots for Windows Temp

steverweber committed:
Merge branch 'develop' into dev_autodoc

Commit:
Merge pull request #42018 from cloudflare/out-exec-mod

Out execution module

terminalmage committed:
Avoid race condition when iterating through processes

If a process has already exited between when psutil.process_iter() was
invoked, and when we invoke the cmdline() func, it will raise an
exception. This commit catches this exception and skips these processes.

Commit:
Add integration test for salt.cloud.CloudClient.create() function

Commit:
Add the 'provider' key to salt.cloud.CloudClient.create()

Much like how the 'profile' key exists in this function, the
'provider' key must also exist for other salt-cloud functions
to work properly. Various salt-cloud functions rely on provider
being available (as it is always available when using cloud profiles),
and so it should be available when no profile is used as well.

Also adjusted some rst syntax in the cloud config file as well
as the documentation around the create() function. Passing in
kwargs as the docs originally suggested does not work.

Fixes #41971

Commit:
Merge pull request #42009 from redmatter/fix-dockerng-network-ignores-test-2017.7

Fix dockerng.network_* ignoring of tests=True

Commit:
Add generate_random_name function to tests helpers

This function is used identically in every cloud provider test file. Let's add
it to a more central place and make it more DRY. Might be useful in other testing
places as well.

Commit:
Replace cloud test __random_name funcs with new generate_random_name helper

Commit:
Replace some more "Nitrogen" references with 2017.7.0

garethgreenaway committed:
Merge pull request #42026 from twangboy/win_unit_test_roots

Fix `unit.fileserver.test_roots` for Windows

Commit:
Merge pull request #42034 from rallytime/fix-41971

Add the 'provider' key to salt.cloud.CloudClient.create()

Commit:
Merge pull request #42032 from terminalmage/py3-tests

Avoid race condition when iterating through processes

Commit:
Merge pull request #42037 from rallytime/nitrogen-doc

Replace some more "Nitrogen" references with 2017.7.0

Commit:
Merge branch '2017.7' into 'develop'

No conflicts.

Commit:
Merge branch '2016.11' into '2017.7'

No conflicts.

jlayze committed:
Update win_service.py

Added support for StartType as kwargs in win_service.enable.  This allows for setting Manual and Auto (Delayed) start types from states.service.running and states.service.dead.

Example state.sls:
"Patch Agent":
  service.running:
    - enable: True
    - StartType: Manual

Commit:
Merge pull request #42035 from rallytime/cloud-test-cleanup

Add generate_random_name function to tests helpers

Commit:
Add mode back in for tmp file creation on Linux

The mode arg was removed in PR #41966 to get this test to pass on
Windows. However, this caused the same test to fail on Arch Linux
when running Python 3.

Let's keep the test the way it was for non-windows boxes, and only
remove the mode for windows.

steverweber committed:
Merge branch 'develop' into dev_autodoc

pgporada committed:
Fixes grammar

jlayze committed:
Update win_service.py

terminalmage committed:
LazyLoader: fix loading module from a subdir

PRs #41937 and #41963 changed the LazyLoader it so that newer Python
releases (>=3.5) usee importlib instead of imp to load modules. However,
when importlib is in use, and the module is being loaded from a
__init__.py, the importlib.machinery.FileFinder instance is created with
the path to the directory containing the __init__.py, which makes it
impossible to find a match for the desired module.

This commit passes the parent dir when instantiating the FileFinder so
that it can properly locate the module.

terminalmage committed:
Add debug logging for grains tests

The assert doesn't tell us anything about the grain that is failing.
This adds some debug logging to make this easier to troubleshoot.

terminalmage committed:
Use existing fpath_dirname variable

This saves a call to os.path.dirname()

Commit:
Merge pull request #42057 from rallytime/ssh-unit-test

Add mode back in for tmp file creation on Linux for ssh unit test

Commit:
Merge pull request #42051 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Merge pull request #42050 from rallytime/merge-2017.7

[2017.7] Merge forward from 2016.11 to 2017.7

Commit:
Merge pull request #42020 from terminalmage/toctree-fix

Add the new docker state modules to the toctree

Commit:
Merge pull request #42011 from steverweber/dev_autodoc

add module and state to help create txt docs from highstate data

Commit:
Merge pull request #42059 from pgporada/fix-grammar

Fixes grammar

jlayze committed:
Update win_service.py

Removed some sneaky tabs

Commit:
Merge pull request #42062 from terminalmage/arch-grains-test

Add debug logging for grains tests

Commit:
Merge pull request #42061 from terminalmage/fix-loader-subdir

LazyLoader: fix loading module from a subdir

Commit:
Merge pull request #42025 from terminalmage/move-mkstemp

Move salt.utils.files.mkstemp temporarily back to salt.utils

terminalmage committed:
salt-ssh: get thin_dir from master_opts if not in opts

Commit:
Merge pull request #42066 from terminalmage/thin-dir-py3

salt-ssh: get thin_dir from master_opts if not in opts

garethgreenaway committed:
When building up the remote file location nothing was being returned via _thin_dir so the remote path was starting with None.  This change updates _thin_dir to return a value for format to use.

glynnforrest committed:
Add note to system module about wrapper scripts preventing shutdown

Fixes #42010

terminalmage committed:
Fix usage of assert_called_once in Python 3 < 3.6

This is a more complete fix which builds upon 41c65b0 and covers all
refs to assert_called_once in the suite.

Commit:
Added decorator jinja_global to add functions as jinja globals

Commit:
Added jinja_raise as 'raise' to be able to raise errors from jinja

Commit:
Added loading of decorated jinja_global functions

Commit:
Merge pull request #42071 from garethgreenaway/42070_state_runs_via_salt_ssh

fix to ssh/wrapper/state.sls

Commit:
Merge branch '2017.7' into 'develop'

No conflicts.

terminalmage committed:
Update mocking in network tests

017fbdb changed the function calls in such a way that tests in 2017.7
now fail. This updates the mocking to account for new code paths
introduced in this commit.

Commit:
Merge pull request #42073 from glynnforrest/system-molly-guard-doc

Add note to system module about wrapper scripts preventing shutdown

gtmanfred committed:
pass cachedir from kwargs in cache

twangboy committed:
Add missing batch file

twangboy committed:
Fix comments

jlayze committed:
Update win_service.py

Added .lower() to the IF statement.

terminalmage committed:
Prevent command from showing in exception when output_loglevel=quiet

When a command is being executed using salt.utils.timed_subprocess.TimedProc,
and that command does not exist in the PATH, it will result in an
OSError/IOError being raised. We catch this and then raise a
CommandExecutionError with some information, including the command being
run. However, when output_loglevel=quiet, we don't want any details of
the command to be visible.

This commit redacts the command details in these corner cases.

twangboy committed:
Use default salt configs for Windows installs

Commit:
Merge pull request #42090 from twangboy/win_add_salt_bat

Add missing batch file

Commit:
Merge pull request #42089 from gtmanfred/spm

pass cachedir from kwargs in cache

twangboy committed:
Create Master installer if script is found

twangboy committed:
Change repo_ng to repo-ng

gtmanfred committed:
require large timediff for ipv6 warning

Commit:
Merge pull request #41953 from yagnik/develop

Fix loader to point to custom loader as opposed to base loader

Commit:
Merge pull request #42096 from twangboy/win_use_default_configs

Use default salt configs for Windows installs

Commit:
Merge pull request #42098 from twangboy/win_repo_docs

Change repo_ng to repo-ng

Commit:
[PY3] Fix haproxyconn unit tests that are failing

``assertItemsEqual`` does not exist in Python 3. It was removed. Since
``assertItemsEqual(a, b)`` is the same thing as asserting that
``sorted(a) == sorted(b)``, we can just update the test in this manner
to work in both Python 2 and 3.

Commit:
Import datetime

Commit:
[Py3] Use functools.cmp_to_key with sorted instead of cmp kwarg

In Python3, using the ``cmp=x`` functionality with sorted is no longer
supported. This older-style comparison function can be replaced in
Python 3, however, with using functools.cmp_to_key as a key argument
instead.

jlayze committed:
Update test_win_service.py

Added StartTypeDelayed to mock_info return for test_enable.

jlayze committed:
Update test_win_service.py

Commit:
Merge pull request #42097 from gtmanfred/timediff

require large timediff for ipv6 warning

Commit:
Added Errors section to jinja doc

Commit:
Update aws.rst - add Debian default username

Commit:
[doc] Update aws.rst : typo fix

Commit:
Merge pull request #42110 from arthurlogilab/patch-21

[doc] Update aws.rst : typo fix

terminalmage committed:
Fix regression when CLI pillar override is used with salt-call

jlayze committed:
Merge pull request #1 from saltstack/develop

Update branch to latest

Commit:
Merge pull request #42084 from terminalmage/assert-called-once

Fix usage of assert_called_once in Python 3 < 3.6

jlayze committed:
Merge pull request #2 from jlayze/develop

Merge latest develop branch to patch-4

gtmanfred committed:
Update salt-master first before updating salt-minion

Fixes #42124

terminalmage committed:
Add integration test for 42116

gtmanfred committed:
add release note about about extmod blacklist and whitelist

gtmanfred committed:
add note about roster_defaults

gtmanfred committed:
fix up PR

Commit:
Merge branch 'develop' into haproxyconn-tests

garethgreenaway committed:
Removing code that was previously in place to account for a situation where we backed up the value of the original seconds parameter in the event that the splay paramter was included.  Recent changes made this code no long necessary, so removing it and simply deleting the "_seconds" parameter from the schedule list output.

Commit:
[PY3] Make test_saltmod unit tests less flaky for Python 3

Separates out the comment from the rest of the state return to make this
test more stable on Python 3. This state run in PY3 adds the minion
names to the comment in a random order, so we can't rely on asserting
against this order in the test.

This PR updates the test so that we are still checking for all of the
elements in the comment, but not relying on the order the minions are
added the comment list.

terminalmage committed:
Revert "decode pillar on minion side on receiving"

This reverts commit 6130722df3af82f0da10bb0d8c77db21e41e74c8.

Commit:
Merge pull request #42131 from gtmanfred/release

Add stuff to release notes

Commit:
Refine support for gnupg v2.1 to 2017.7 branch

garethgreenaway committed:
Merge pull request #42139 from terminalmage/revert-37222

Revert "decode pillar on minion side on receiving"

Commit:
Reduce the number of days an issue is stale by 25

This reduces the amount of time an issue is stale to approximately
a little less than 3 years and 2 months.

garethgreenaway committed:
Merge pull request #42134 from terminalmage/pr-42094-2017.7

Prevent command from showing in exception when output_loglevel=quiet (2017.7 branch)

garethgreenaway committed:
Merge pull request #42130 from terminalmage/issue42116-2017.7

Fix regression in CLI pillar override for salt-call (2017.7 branch)

Commit:
Merge pull request #42133 from garethgreenaway/42127_schedule_always_trigger_change_fix

2017.7.0 modules/schedule.py fix

Commit:
Added wraps decorated so sys.doc works on decorated functions

Commit:
Fix to get_service_instance_via_proxy

garethgreenaway committed:
Merge pull request #42147 from dmurphy18/fix_gnupgv21_for_20177

Refine support for gnupg v2.1 for 2017. 7 branch

Commit:
Issue 42162 - Route53 boto module should retry on PriorRequestNotComplete error

Commit:
Merge pull request #42085 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Merge branch 'develop' into haproxyconn-tests

Commit:
delete temp file

Commit:
Merge pull request #42149 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 25

Commit:
Merge pull request #42138 from rallytime/py3-saltmod-tests

[PY3] Make test_saltmod unit tests less flaky for Python 3

Commit:
Merge pull request #42109 from arthurlogilab/patch-20

[doc] Update aws.rst - add Debian default username

Commit:
Merge pull request #42102 from rallytime/py3-kernelpkg-tests

[Py3] Use functools.cmp_to_key with sorted instead of cmp kwarg

Commit:
Merge pull request #42101 from rallytime/haproxyconn-tests

[PY3] Fix haproxyconn unit tests that are failing

Commit:
Merge pull request #42022 from terminalmage/issue41996

salt.modules.dockermod: ensure that json.loads is fed a string

terminalmage committed:
importlib: Use load_module() instead of exec_module() to initialize loaded mods

Using exec_module() results in some unexpected behavior in the LazyLoader,
so more investigation will need to be done before we can use it. In the
meantime, this commit uses load_module() instead. It also gets rid of
the now-unnecessary workarounds we had recently instituted in salt-ssh
to work around issues with the __opts__ dunder.

jlayze committed:
Merge branch 'develop' into patch-4

garethgreenaway committed:
Merge pull request #42088 from terminalmage/update-mocking

Update mocking in network tests

garethgreenaway committed:
Merge pull request #42184 from terminalmage/py3-tests

importlib: Use load_module() instead of exec_module() to initialize loaded mods

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - salt/grains/core.py

Commit:
add destroying and destroy cloud events to azurearm provider

amontalban committed:
Fix Augeas state documentation example

jlayze committed:
Add delay option to beacons/service.py

terminalmage committed:
Fix another instance where json.loads needs a string

jlayze committed:
Update service.py

Fixed no return if no delay. (Lines 125-126)

Commit:
Convert remaining "assert_called_once" references to new syntax

These tests are new to develop and need to be converted to use the new
assert_called_once syntax introduced in #42084.

Commit:
Merge pull request #42053 from jlayze/patch-4

Update win_service.py

gtmanfred committed:
allow not verifying ssl winrm saltcloud

Required for Self Signed Certs on pywinrm>=0.2.1

gtmanfred committed:
add document for winrm_verify_ssl

gtmanfred committed:
add release note

garethgreenaway committed:
Merge pull request #42208 from terminalmage/ensure-json-string

Fix another instance where json.loads needs a string

jlayze committed:
Update service.py

Fixed errors from salt-pr-lint-n test.

Commit:
Merge pull request #42192 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Lint: Remove extra spaces between `name` and `}`

Commit:
Merge pull request #42201 from shengis/add_cloud_event_for_azurearm

add destroying and destroy cloud events to azurearm provider

tdutrion committed:
Remove duplicate instruction in Openstack Rackspace config example

Commit:
add custom dns support for azurearm provider
- added option in documentation : dns_servers

Commit:
spaces removed for lint

Ch3LL committed:
[2017.7] parse libcloud rc version correctly

Commit:
Merge pull request #42224 from tdutrion/develop

Remove duplicate instruction in Openstack Rackspace config example

Commit:
Merge branch 'develop' into patch-6

garethgreenaway committed:
Merge pull request #42234 from Ch3LL/fix_cloud_rc

[2017.7] parse libcloud rc version correctly

Commit:
fix using isinstance

Commit:
Merge branch 'develop' into add_dns_for_azurearm

alexbleotu committed:
Merge branch 'develop' into jinja_globals_gh

gtmanfred committed:
fix signature versionadded release

Commit:
Merge pull request #42229 from shengis/add_dns_for_azurearm

add custom dns support for azurearm provider

Commit:
Add get_driver function used with libcloud modules to doc exception list

Fixes failing valid_docs test in develop

garethgreenaway committed:
Merge branch 'develop' into patch-6

gtmanfred committed:
Merge branch 'develop' into gpg

garethgreenaway committed:
Merge pull request #42249 from rallytime/docstring-fix

Add get_driver function used with libcloud modules to doc exception list

jasperla committed:
Extend resolution of 'virtual' grain for OpenBSD on KVM.

Commit:
Merge pull request #42248 from gtmanfred/gpg

fix signature versionadded release

Commit:
Merge branch 'develop' into issue_42162_route53boto

Commit:
Merge branch '2017.7' into 'develop'

Conflicts:
  - salt/cloud/libcloudfuncs.py

Commit:
Merge pull request #42214 from gtmanfred/develop

allow not verifying ssl winrm saltcloud

CorvinM committed:
Add batch_wait support for Python API

Fixes #42225

Commit:
Merge branch 'develop' into jinja_globals_gh

Commit:
Merge branch 'develop' into decorator_fixes-gh

jasperla committed:
Merge branch 'develop' into openbsd_kvm

Commit:
fix typo

Commit:
Deprecation warning for retry_on_rate_limit and rate_limit_retries args

Commit:
Merge remote-tracking branch 'upstream/develop' into issue_42162_route53boto

Commit:
Merge branch 'issue_42162_route53boto' of https://github.com/fpires82/salt into issue_42162_route53boto

Commit:
Warning should be until Neon, not Oxygen

Commit:
Merge pull request #42271 from CorvinM/issue42225

Add batch_wait support for Python API

Commit:
Merge pull request #42268 from rallytime/merge-develop

[develop] Merge forward from 2017.7 to develop

Commit:
Merge pull request #42281 from SaltyCharles/patch-7

fix typo

mirceaulinic committed:
test.get_opts runner function, similar to the exec fun

Commit:
Fix index error when setting empty values

When setting empty values, the string returned by sysrc would be split in a way that would make the result a list containing just the delimiter " ->". This would then produce a failure when trying to select the second entry in the list.

jlayze committed:
Merge branch 'develop' into patch-6

Commit:
Fix bug in github.absent state which can cause some users to not be removed from an organization.

Commit:
Remove sock_dir deprecation warning from salt.utils.cloud.fire_event

`sock_dir` was changed from an optional kwarg to a required arg, the
warning was removed, and all instances of `cloud.fire_event` were
examined to make sure there were no missing `sock_dir` args. I found
a couple of references where `sock_dir` was missing, so I added it
in those places.

Commit:
Remove Oxygen deprecation warning in lxc execution module

Commit:
Add this change to the Oxygen release notes

Commit:
Merge pull request #42299 from ecdavis/github_fix-casing-bug-in-absent

Fix bug in github.absent state which can cause some users to not be removed from an organization.

Commit:
Merge pull request #42256 from jasperla/openbsd_kvm

Extend resolution of 'virtual' grain for OpenBSD on KVM.

Commit:
Merge pull request #42171 from fedepires/issue_42162_route53boto

Issue #42162 - Route53 boto module should retry on PriorRequestNotComplete

layer3switch committed:
aws metadata string prefix match tuple added

Prefix path ending with `latest/dynamic` and `latest/meta-data` should be handled differently as it only returns body content for those paths. Also due to the `/` path handling returns last element in the array, secondary conditional is added.

Commit:
Merge pull request #42303 from rallytime/lxc-oxygen-removal

Remove Oxygen deprecation warning in lxc execution module

Commit:
Merge pull request #42305 from layer3switch/patch-1

AWS specific metadata string prefix tuple match added

Commit:
Reduce the number of days an issue is stale by 15

We're still within the 3 years and 2 months range. I opted to only
reduce this number by 15 instead of 25 because this change will mark
18 issues as stale rather than about 35.

We need to keep reducing this number, but we also don't want to flood
this process too much.

garethgreenaway committed:
Merge branch 'develop' into patch-6

garethgreenaway committed:
Merge pull request #42313 from rallytime/stale-config-update

Reduce the number of days an issue is stale by 15

luk4hn committed:
fix #42315, add option to disable to use of security groups

In advanced zones, security groups are supported only on the KVM hypervisor.
With hypervisor other than KVM, we need a way to disable the use of SGs

ref: https://svn.apache.org/repos/asf/cloudstack/docsite/html/docs/en-US/Apache_CloudStack/4.1.0/html/Admin_Guide/security-groups.html

Commit:
fix the doc template

Commit:
also return the error from lvcreate

luk4hn committed:
add option to disable to use of security groups (fix lintint)

asyncmind0 committed:
Check remote tags before deciding to do a fetch #42329

salt-jenkins commented:
@jagguli, thanks for your PR! By analyzing the history of the files in this pull request, we identified @terminalmage, @robgott and @cheuschober to be potential reviewers.

asyncmind0 committed:
Handle forward compatibility for azure arm

huynqtiki committed:
assign get_security_groups to a variable so it isn't called twice

not-raspberry committed:
Document how to specify view permissions

Example:

`pg_stat_database` is a view in `pg_catalog`.

```
datadog_grants:
  postgres_privileges.present:
    - name: datadog
    - prepend: pg_catalog
    - object_name: pg_stat_database
    - object_type: table
    - privileges:
      - SELECT
```

cachedout commented:
@terminalmage Could you please have a look here? Thanks.

Commit:
Merge pull request #42326 from steverweber/lvm_state_error_help

lvm state return more helpful error on lvcreate

Commit:
Merge pull request #42325 from steverweber/auto_doc_fix

fix the highstate_doc template

Commit:
Merge pull request #42321 from luk4hn/fix-42315

fix #42315, add option to disable the use of security groups

Commit:
Merge pull request #42302 from rallytime/cloud-utils-fire-event-deprecation

Remove sock_dir deprecation warning from salt.utils.cloud.fire_event

Commit:
Merge pull request #42298 from schasi/patch-1

Fix index error when setting empty values

Commit:
Merge pull request #42332 from StreetHawkInc/fix_azurearm

Handle forward compatibility for azurearm

Commit:
Merge pull request #42296 from not-raspberry/patch-1

Document how to specify view permissions

Commit:
Merge pull request #42287 from cloudflare/runner-get-opts

test.get_opts runner function, similar to the exec fun

Commit:
Merge pull request #42245 from alexbleotu/decorator_fixes-gh

Minor vsphere fixes

Commit:
Merge pull request #42244 from alexbleotu/jinja_globals_gh

Jinja raise function and globals decorator

Commit:
Merge pull request #42203 from amontalban/augeas_doc_fix

Fix Augeas state documentation example

Commit:
Merge pull request #42207 from jlayze/patch-6

Add delay option to beacons/service.py

asyncmind0 committed:
Merge branch 'develop' into develop

terminalmage reviewed (changes requested):
This PR has gotten kinda messed up with the changes to the head branch. You may need to cherry-pick your changes into a branch that is branched off of 2016.11 and then open a new PR.
